compiler/hash-tir/src/pats.rs

```diff
@@ -33,32 +33,55 @@ pub struct Spread {
     pub index: usize,
 }
 
-/// A range pattern containing two bounds `start` and `end`.
+/// A range pattern containing two bounds `lo` and `hi`.
 ///
-/// The `start` and `end` values must be either both [`LitPat::Int`], or both
+/// The `lo` and `hi` values must be either both [`LitPat::Int`], or both
 /// [`LitPat::Char`].
 #[derive(Copy, Clone, Debug)]
 pub struct RangePat {
     /// The beginning of the range.
-    pub start: LitPat,
+    pub lo: LitPat,
+
     /// The end of the range.
-    pub end: LitPat,
+    pub hi: LitPat,
+
     /// If the range includes the `end` or not.
-    pub range_end: RangeEnd,
+    pub end: RangeEnd,
 }
 
 /// Represents a pattern.
 ///
 /// Check the documentation of each member for more information.
 #[derive(Copy, Clone, Debug, From)]
 pub enum Pat {
```

Comment:
Duplicate documentation with the member structs?

Reply:
Which part do you mean?

Reply:
Indeed, there is a bit of duplication. I still think it's useful to summarise all of the content in one place, but it can be removed.

Reply:
Maybe just a shortened version? otherwise they will eventually start getting out of sync